### PR TITLE
fix(highlight): add `StatusLineTerm`/`StatusLineTermNC` to `:color vim`

### PR DESCRIPTION
**Pull request by echasnovski**
Problem: both `StatusLineTerm`/`StatusLineTermNC` are now explicitly
  used, but `:color vim` does not set them to the values used in Vim.
  This might be fine if `:color vim` is treated as "the state of default
  color scheme prior the big update", but it seems to be better treated
  as "Vim's default color scheme" (how it is documented in its header).

Solution: add `StatusLineTerm`/`StatusLineTermNC` definitions to
  'runtime/colors/vim.lua'.
  Use explicit foreground colors ('Whte'/'Black') instead of `guifg=bg`
  used in source, as the latter caused some problems in the past (if
  `Normal` is not defined, `nvim_set_hl()` can't recognize `'bg'` as the
  foreground value).
  Also realign the rest of the background conditional highlight groups.

------

Also update some functional tests to make them pass.

------

Follow up to #29290.
[Original context](https://github.com/neovim/neovim/pull/29290#issuecomment-2164862061).
Vim's references: [light background](https://github.com/vim/vim/blob/02f3ebacfbfa1f892347d7532278f24620e68300/src/highlight.c#L344-L347), [dark background](https://github.com/vim/vim/blob/02f3ebacfbfa1f892347d7532278f24620e68300/src/highlight.c#L433-L436).

### CODE REVIEW

**Comment by zeertzjq** [comment]
Needs rebase on #29326

**Comment by echasnovski** [comment]
> Needs rebase on #29326

Done.

File: test/functional/terminal/tui_spec.lua [hunk 1]
```diff
@@ -1552,6 +1557,29 @@ describe('TUI', function()
       [3] = { { bold = true }, { bold = true } },
       [4] = { { fg_indexed = true, foreground = tonumber('0xe0e000') }, { foreground = 3 } },
       [5] = { { foreground = tonumber('0xff8000') }, {} },
+      [6] = {
+        { bold = true, background = Screen.colors.LightGreen, foreground = Screen.colors.Black },
+        { bold = true },
+      },
```
Comment (zeertzjq) [inline test/functional/terminal/tui_spec.lua:1563]: Attribute ID 2 is now unused, so you can replace attribute 2 with this.

File: test/functional/terminal/tui_spec.lua [hunk 2]
```diff
@@ -1056,14 +1056,19 @@ describe('TUI', function()
     if is_ci('github') then
       pending('tty-test complains about not owning the terminal -- actions/runner#241')
     end
+    screen:set_default_attr_ids({
+      [1] = { reverse = true }, -- focused cursor
+      [3] = { bold = true },
+      [19] = { bold = true, background = 121, foreground = Screen.colors.Black }, -- StatusLineTerm
```
Comment (zeertzjq) [inline test/functional/terminal/tui_spec.lua:1062]:
```suggestion
      [19] = { bold = true, background = 121, 0 }, -- StatusLineTerm
```

File: test/functional/terminal/tui_spec.lua [hunk 3]
```diff
@@ -1552,6 +1557,29 @@ describe('TUI', function()
...
+      },
+      [7] = {
+        {
+          fg_indexed = true,
+          bg_indexed = true,
+          bold = true,
+          background = tonumber('0x66ff99'),
+          foreground = Screen.colors.Black,
+        },
+        { bold = true, background = 121, foreground = Screen.colors.Black },
```
Comment (zeertzjq) [inline test/functional/terminal/tui_spec.lua:1572]:
```suggestion
        { bold = true, background = 121, foreground = 0 },
```

File: test/functional/terminal/tui_spec.lua [hunk 4]
```diff
@@ -1552,6 +1557,29 @@ describe('TUI', function()
...
+      },
+      [8] = {
+        {
+          fg_indexed = true,
+          bg_indexed = true,
+          background = tonumber('0x66ff99'),
+          foreground = Screen.colors.Black,
+        },
+        { background = 121, foreground = Screen.colors.Black },
```
Comment (zeertzjq) [inline test/functional/terminal/tui_spec.lua:1581]:
```suggestion
        { background = 121, foreground = 0 },
```

File: test/functional/ui/wildmode_spec.lua [hunk 5]
```diff
@@ -199,9 +199,13 @@ describe("'wildmenu'", function()
     feed((':terminal "%s" REP 5000 !terminal_output!<cr>'):format(testprg('shell-test')))
     feed('G') -- Follow :terminal output.
     feed([[:sign <Tab>]]) -- Invoke wildmenu.
+    screen:set_default_attr_ids {
+      [31] = { foreground = Screen.colors.Grey0, background = Screen.colors.Yellow },
```
Comment (zeertzjq) [inline test/functional/ui/wildmode_spec.lua:203]:
```suggestion
      [31] = { foreground = Screen.colors.Black, background = Screen.colors.Yellow },
```

File: test/functional/ui/wildmode_spec.lua [hunk 6]
```diff
@@ -259,9 +263,13 @@ describe("'wildmenu'", function()
 
     feed([[<C-\><C-N>]])
     feed([[:<Tab>]]) -- Invoke wildmenu.
+    screen:set_default_attr_ids {
+      [31] = { foreground = Screen.colors.Grey0, background = Screen.colors.Yellow },
```
Comment (zeertzjq) [inline test/functional/ui/wildmode_spec.lua:267]:
```suggestion
      [31] = { foreground = Screen.colors.Black, background = Screen.colors.Yellow },
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**